base/expr.jl

```diff
@@ -67,8 +67,9 @@ function copy(c::CodeInfo)
     cnew.slotflags = copy(cnew.slotflags)
     cnew.codelocs  = copy(cnew.codelocs)
     cnew.linetable = copy(cnew.linetable)
-    cnew.ssaflags = copy(cnew.ssaflags)
-    ssavaluetypes = cnew.ssavaluetypes
+    cnew.ssaflags  = copy(cnew.ssaflags)
+    cnew.edges     = cnew.edges === nothing ? nothing : copy(cnew.edges)
```

Comment:
@JeffBezanson: this is why it's sometimes handy to have `copy` methods for immutables—you want to copy something in case it's mutable, but maybe it isn't.

Reply:
In general you need `x === nothing ? nothing : f(x)`; this is just a case of that. This argument clearly wouldn't apply to functions other than `copy`; I suspect `copy` just feels like a more reasonable function to implement since it's so close to the identity function.